tensorflow_datasets/core/dataset_builder.py

```diff
@@ -1040,9 +1115,11 @@ class GeneratorBasedBuilder(FileAdapterBuilder):
   `tensorflow_datasets.core.file_format_adapter` and specify constraints on the
   feature dictionaries yielded by example generators. See the class docstrings.
   """
+  _version: utils.Version
+  _data_dir: str
 
   @abc.abstractmethod
-  def _generate_examples(self, **kwargs):
+  def _generate_examples(self, **kwargs: Dict[Any, Any]) -> Iterable:
```

Comment:
* `Iterator[MyType]` and similarly everywhere else `Iterator` used.
* You should not specify the `Dict` for kwargs annotation, so it should be `**kwargs: Any`

Reply:
Done